main/SAPI.h

```diff
@@ -271,7 +275,7 @@ struct _sapi_post_entry {
 	char *content_type;
 	uint32_t content_type_len;
 	void (*post_reader)(void);
-	void (*post_handler)(char *content_type_dup, void *arg);
+	void (*post_handler)(char *content_type_dup, zval *arg, zval *files);
```

Comment:
This requires update in all SAPIs. Can we avoid this?

Reply:
I suppose we can prevent these changes by doing some dirty copies of $_POST/$_FILES from request_parse_body, using the new values as return values for the function and restoring the old values. If you prefer that then I'll adjust the code accordingly.

Reply:
Why would this require update of all SAPIs? This is just for sapi_register_post_entries which has got only one core user (mbstring). I don't see an issue of updating this in master as ABI breaks are fine and this is not really used much IMHO.

Reply:
@bukka You are right. I thought this is a change in ``_sapi_module_struct``.

Reply:
I don't object against this patch, but it would be great to clean-up it a bit.